pkg/exporter/exporter.go

```diff
@@ -0,0 +1,24 @@
+package exporter
+
+// Type defines Exporter Type
+type Type int
+
+const (
+	// AzureBlob defines AzureBlob exporter Type
+	AzureBlob Type = iota
+)
+
+// Name returns type name
+func (t Type) name() string {
+	return [...]string{"azureblob"}[t]
+}
+
+// BaseExporter defines Base Exporter
+type BaseExporter struct {
+	exporterType Type
+}
+
+// GetName gets exporter name
+func (b *BaseExporter) GetName() string {
+	return b.exporterType.name()
+}
```

Comment:
make exporter design consistent with collector and diagnoser as there is no reason for it to be different. This allows new exporters to be defined using the same pattern as collector / diagnose